handle_multi_pay_invoice_request.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/nbd-wtf/go-nostr"
+	decodepay "github.com/nbd-wtf/ln-decodepay"
+	"github.com/sirupsen/logrus"
+)
+
+func (svc *Service) HandleMultiPayInvoiceEvent(ctx context.Context, sub *nostr.Subscription, request *Nip47Request, event *nostr.Event, app App, ss []byte) {
+
+	nostrEvent := NostrEvent{App: app, NostrId: event.ID, Content: event.Content, State: "received"}
+	err := svc.db.Create(&nostrEvent).Error
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+			"appId":     app.ID,
+		}).Errorf("Failed to save nostr event: %v", err)
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+		}).Errorf("Failed to process event: %v", err)
+		return
+	}
+
+	multiPayParams := &Nip47MultiPayParams{}
+	err = json.Unmarshal(request.Params, multiPayParams)
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+			"appId":     app.ID,
+		}).Errorf("Failed to decode nostr event: %v", err)
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+		}).Errorf("Failed to process event: %v", err)
+		return
+	}
+
+	var wg sync.WaitGroup
+	for _, invoiceInfo := range multiPayParams.Invoices {
+		wg.Add(1)
+		go func(invoiceInfo InvoiceInfo) {
+			defer wg.Done()
+			bolt11 := invoiceInfo.Invoice
+			// Convert invoice to lowercase string
+			bolt11 = strings.ToLower(bolt11)
+			paymentRequest, err := decodepay.Decodepay(bolt11)
+			if err != nil {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":   event.ID,
+					"eventKind": event.Kind,
+					"appId":     app.ID,
+					"bolt11":    bolt11,
+				}).Errorf("Failed to decode bolt11 invoice: %v", err)
+
+				resp, err := svc.createResponse(event, Nip47Response{
+					ResultType: NIP_47_PAY_INVOICE_METHOD,
+					Error: &Nip47Error{
+						Code:    NIP_47_ERROR_INTERNAL,
+						Message: fmt.Sprintf("Failed to decode bolt11 invoice: %s", err.Error()),
+					},
+				}, ss)
+				if err != nil {
+					svc.Logger.WithFields(logrus.Fields{
+						"eventId":        event.ID,
+						"eventKind":      event.Kind,
+						"paymentRequest": invoiceInfo.Invoice,
+						"invoiceId":      invoiceInfo.Id,
+					}).Errorf("Failed to process event: %v", err)
+					return
+				}
+				dTag := []string{"a", fmt.Sprintf("%d:%s:%d", NIP_47_RESPONSE_KIND, event.PubKey, invoiceInfo.Id)}
```

Comment:
I don't understand this d tag, what is the value? it doesn't seem to match the spec.

Also you construct it in 4 different places, can't you just do it once?

Reply:
Yeah that's also why `createAndAppendResponse` existed.

> Then we should also remove createAndAppendResponse and not deal with returning an array of responses.

But I thought you weren't happy with it

Reply:
> I don't understand this d tag, what is the value? it doesn't seem to match the spec.

I thought d tag [here](https://github.com/nostr-protocol/nips/pull/685/files#diff-d1ed4c38da272aa7f99e29edb6739e8fb209c723585601a84f0fd511d1c679e0R220-R222) meant [this](https://github.com/nostr-protocol/nips/blob/master/01.md?plain=1#L82), am I wrong?

Reply:
> I thought d tag [here](https://github.com/nostr-protocol/nips/pull/685/files#diff-d1ed4c38da272aa7f99e29edb6739e8fb209c723585601a84f0fd511d1c679e0R220-R222) meant [this](https://github.com/nostr-protocol/nips/blob/master/01.md?plain=1#L82), am I wrong?
I think it should be a simple tag like `["d", "ID_OR_PAYMENT_HASH HERE"]`,

> Yeah that's also why createAndAppendResponse existed.
I mean, can't you create a variable to store the d tag at the start of the loop and then use it wherever you need it, rather than declare it 4 times? just to keep it DRY (or have a function that does it, but `createAndAppendResponse` did something different, right? we do not append a response any more)

Reply:
> Yeah that's also why createAndAppendResponse existed.

Sorry ^one of the reasons why

But changed with the new commit now (although could only dry up dTag declarations)

Reply:
I also use `d` tags in the way I mentioned in the [PoS app](https://github.com/getAlby/pos/blob/master/src/pages/wallet/Profile.tsx#L44). I will fix this.